src/CMakeLists.txt

```diff
@@ -1,6 +1,12 @@
 configure_file ( version.h.in version.h ESCAPE_QUOTES @ONLY )
 
-target_sources ( ${PROJECT_NAME} PRIVATE main.cpp server.cpp packet.cpp attribute.cpp vendor_attribute.cpp utils.cpp dictionaries.cpp)
+add_library ( radproto STATIC )
+
+target_sources ( radproto PRIVATE server.cpp packet.cpp attribute.cpp vendor_attribute.cpp utils.cpp dictionaries.cpp)
+
+target_link_libraries ( ${PROJECT_NAME} radproto )
+
+target_include_directories ( radproto PUBLIC ${CMAKE_CURRENT_SOURCE_DIR} )
 
 target_link_libraries ( ${PROJECT_NAME} OpenSSL::Crypto Boost::boost Threads::Threads )
```

Comment:
This must go to the sample server.

Reply:
However you also need these imported targets in `radproto` library. Linkage does not happen here, but imported targets also provide include directories and compiler flags (`Threads::Threads' provide `-pthread` flag). Compilation may fail if OpenSSL or Boost are installed in non-standard places.

Reply:
There should be no mentions of the sample server in this CMakeLists.txt (`${PROJECT_NAME}`). This statement should relate to the library.